### PR TITLE
tmux: Add default terminal to fix backspaces

### PR DESCRIPTION
**Pull request by TheRealKeto**
#### Description

Without specifying a default terminal while configuring, tmux fails to correctly display itself when backspacing.

Closes: https://trac.macports.org/ticket/65923

###### Type(s)

- [x] bugfix

###### Tested on

macOS 12.6 21G115 arm64
Xcode 14.2 14C18

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Review by tessus — APPROVED** [review]
LGTM

**Review by reneeotten — CHANGES_REQUESTED** [review]
Functional- and non-functional changes (i.e., whitespace in the Portfile) should not be mixed in a single commit. 

Please separate those in two if there is a strong need to make the non-functional changes; it seems the Portfile was already aligned with four spaces it seems unnecessary. You can always use line-continuation to wrap something on the next line.

File: sysutils/tmux/Portfile [hunk 1]
```diff
@@ -83,6 +86,4 @@ post-destroot {
     xinstall -m 0644 ${filespath}/ftdetect-tmux.vim ${destroot}${prefix}/share/vim/vimfiles/ftdetect/tmux.vim
 }
 
-notes "If you want integration with system pasteboard consider installing port tmux-pasteboard as well"
-
```
Comment (reneeotten) [inline sysutils/tmux/Portfile:87]: did you intend to remove the notes? It does appear to me that this is unrelated to the commit message, and the notes look still relevant... or?

Comment (TheRealKeto) [reply sysutils/tmux/Portfile:87]: The trac issue linked is addressed by the commit as well. The [homepage of tmux-pasteboard](https://github.com/ChrisJohnsen/tmux-MacOSX-pasteboard) states that the project isn't needed on newer versions of tmux.
> You may not even need the program from this repository! tmux version 2.6 (2017 September) incorporated the functionality of the wrapper program. If you are using a new enough tmux, you may be able to skip using the wrapper in your tmux configuration.

The note can only address individuals in legacy systems, if possible. That's a good compromise.

File: sysutils/tmux/Portfile [hunk 2]
```diff
@@ -83,6 +86,9 @@ post-destroot {
     xinstall -m 0644 ${filespath}/ftdetect-tmux.vim ${destroot}${prefix}/share/vim/vimfiles/ftdetect/tmux.vim
 }
 
-notes "If you want integration with system pasteboard consider installing port tmux-pasteboard as well"
+notes {
+    On legacy systems, consider installing port tmux-pasteboard to
```
Comment (reneeotten) [inline sysutils/tmux/Portfile:90]: add a line-continuation here so that MacPorts can wrap the line at the terminal width

File: sysutils/tmux/Portfile [hunk 3]
```diff
@@ -32,6 +32,9 @@ license         BSD
 
 depends_lib     port:libevent port:ncurses
 
+# Use screen by default to display tmux correctly when backspacing
+configure.args-append   --with-TERM=screen
```
Comment (reneeotten) [inline sysutils/tmux/Portfile:36]: Since you're changing a configure argument here and the build will differ from the current situation; you should increase the `revision` so that everyone gets the new build. That likely applies to both the `tmux` and `tmux-devel` ports.

**Review by reneeotten — CHANGES_REQUESTED** [review]
sorry @TheRealKeto looking at it again now, I think it needs one final revision...

**Review by reneeotten — APPROVED** [review]

**Comment by reneeotten** [comment]
thanks @TheRealKeto and apologies for the few review rounds...
